### ISSUE TITLE
It seems that playwright isn't installed correctly in docker

### ISSUE THREAD

**Issue by smilefufu** [OpenDevin/OpenDevin#1172]
#### Describe the bug
Opendevin tried to use playwright to browser online, but failed because playwright isn't installed correctly in docker.
`playwright install` command runned, but failed.(see screenshot below)

#### Setup and configuration
**Current version**:

```
I'm using the latest docker image.
```

**My operating system**: MacOS (M1 Max chip)

**My environment vars and other configuration** (be sure to redact API keys):
```bash
export WORKSPACE_DIR=/Users/fufu/codes/playgruond/workspace4opendevin/test_project
export LLM_API_KEY="sk-non-important"
export LLM_BASE_URL="https://someproxydomain2openai.com/v1"
```

**My model and agent** (you can see these settings in the UI):
* Model: GPT3.5
* Agent: MonologueAgent

**Commands I ran to install and run OpenDevin**:
```bash
docker run \
    -e LLM_API_KEY=$LLM_API_KEY \
    -e LLM_BASE_URL=$LLM_BASE_URL \
    -e WORKSPACE_MOUNT_PATH=$WORKSPACE_DIR \
    -v $WORKSPACE_DIR:/opt/workspace_base \
    -v /var/run/docker.sock:/var/run/docker.sock \
    -p 3000:3000 \
    ghcr.io/opendevin/opendevin:latest
```

**Steps to Reproduce**:
1. Create a pixel style RPG game.
2.
3.

**Logs, error messages, and screenshots**:

```
ACTION:
BrowseURLAction(url='https://www.pixelart.com/', action=<ActionType.BROWSE: 'browse'>)
huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...
To disable this warning, you can either:
	- Avoid using `tokenizers` before the fork if possible
	- Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)

OBSERVATION:
BrowserType.launch: Executable doesn't exist at /root/.cache/ms-playwright/chromium-1112/chrome-linux/chrome
╔════════════════════════════════════════════════════════════╗
║ Looks like Playwright was just installed or updated.       ║
║ Please run the following command to download new browsers: ║
║                                                            ║
║     playwright install                                     ║
║                                                            ║
║ <3 Playwright Team                                         ║
╚════════════════════════════════════════════════════════════╝

```

<img width="1265" alt="image" src="https://github.com/OpenDevin/OpenDevin/assets/4025427/e40e105f-19e9-4e74-b3d3-9ec350025291">

**Comment by smilefufu** [comment]
I think this is caused by PATH env didn't set correctly in docker. Maybe try `python -m playwright install` ?

**Comment by smilefufu** [comment]
I think I've solved the problem by running command `playwright install-deps` mannually in `docker exec` in docker (opendevin:latest)
<img width="820" alt="image" src="https://github.com/OpenDevin/OpenDevin/assets/4025427/dc94ad1b-3170-4cb6-98aa-5c196f3a16a4">

**Comment by rbren** [comment]
Good catch! We probably just need to add that command here: https://github.com/OpenDevin/OpenDevin/blob/main/containers/app/Dockerfile

**Comment by Umpire2018** [comment]
![image](https://github.com/OpenDevin/OpenDevin/assets/138990495/ff9e8b86-ffe0-4d65-8d52-d5a08a33f59c)

Please add `RUN npx -y playwright@1.43.0 install --with-deps`  to line `53` in `containers/app/Dockerfile` then rebuild image to help check if this fix the problem.

**Comment by SmartManoj** [comment]
@Umpire2018 I think that's for interacting using node.

@rben after 27?https://github.com/OpenDevin/OpenDevin/blob/aed82704a947df6e348a5fcf044c482f8e580f5f/containers/app/Dockerfile#L27

**Comment by Umpire2018** [comment]
> @Umpire2018 I think that's for interacting using node.

<img width="767" alt="image" src="https://github.com/OpenDevin/OpenDevin/assets/138990495/90c1c93a-27e4-4c22-9b83-514f6e9a94fe">

I believe it's to install `playwright` system-wide.  Below is its suggestion for your reference.

<img width="587" alt="image" src="https://github.com/OpenDevin/OpenDevin/assets/138990495/666db28f-0ae8-4657-b1fc-79e989ce5933">

**Comment by rbren** [comment]
Most likely both `python` and `npm` can be used to install playwright globally. Since we're calling playwright from python, we should probably use that one.

@SmartManoj yeah I think you found the best spot for it

**Comment by foragerr** [comment]
In the original issue at the top, the agent is trying to access Playwright. Does that not mean Playwright needs to be installed on the sandbox image rather than the app image?

**Comment by foragerr** [comment]
If it does go in the app image, it should be in the runtime stage, so perhaps closer to line # 40

**Comment by SmartManoj** [comment]
BrowseURLAction is happening on the app image.

**Comment by foragerr** [comment]
> The OP installed it in the sandbox itself.
As I understand it, it's the Agent that is trying to install playwright (and failing), rather than the OP.

> BrowseURLAction is happening on the app image.
👍 Someone needs to tell the agent that there's a `BrowseURLAction` it can use instead of installing playwright. Perhaps in the system prompt.

> The command should be run only once. So, it should be in the backend-builder. right?
if you install it in the backend-builder, it wont make it to the final image, unless if you're explicitly copying it over. "runtime" doesn't mean it'll get run everytime the image is launched. see: https://docs.docker.com/build/building/multi-stage/

**Comment by rbren** [comment]
I think this is what happened:
* agent issues a `browse` action
* backend (not sandbox) tries to run playwright
* backend reports an error "playwright not installed" to the agent
* agent tries to run `install playwright` in the sandbox
  * but obviously that won't fix the issue, because it's the _backend_ that needs access to the command

**Comment by SmartManoj** [comment]
> The OP installed it in the sandbox itself.
 
@foragerr  Manually using the docker GUI

**Comment by foragerr** [comment]
Could be, give it a try.

**Comment by rbren** [comment]
can anyone try https://github.com/OpenDevin/OpenDevin/pull/1184 to see if it works?
